### ISSUE TITLE
The diagram is not loading

### ISSUE THREAD

**Issue by CamiAndrei** [xwikisas/application-diagram#82]
Steps to reproduce:
1. Instal Diagram Application version 1.10
2. Get a license
3. Create a Diagram and save 
4. View the diagram

Results
Sometimes the diagram is not loading, on Mozilla Firefox. Spinner icon is displayed (Couldn't force the display with CTRL+F5 ). 
I have reproduced this with 8.4.5 and 9.11.3, but not with 11.8.1 . As long we support the older  version the diagram should work on those too.
This is a regression introduced in 1.9. 
Enviroment:  Linux Mint; Browsers: Mozilla Firefox 69 and Google Chrome 77; XWiki  versions 8.4.5 and 9.11.3, and 11.8.1 on Jetty, hsqldb.
Also in the console I got the following error:  

`[http://xwiki:8080/xwiki/webjars/wiki%3Axwiki/draw.io/12.0.0/styles/grapheditor.css] ERROR ebJarsResourceReferenceHandler - Failed to read resource [draw.io/12.0.0/styles/grapheditor.css] 
org.xwiki.resource.ResourceReferenceHandlerException: Failed to read resource [draw.io/12.0.0/styles/grapheditor.css]
	at org.xwiki.resource.servlet.AbstractServletResourceReferenceHandler.serveResource(AbstractServletResourceReferenceHandler.java:187)
	at org.xwiki.resource.servlet.AbstractServletResourceReferenceHandler.handle(AbstractServletResourceReferenceHandler.java:92)
	at org.xwiki.resource.internal.DefaultResourceReferenceHandlerChain.handleNext(DefaultResourceReferenceHandlerChain.java:79)
	at org.xwiki.resource.internal.AbstractResourceReferenceHandlerManager.handle(AbstractResourceReferenceHandlerManager.java:82)
	at org.xwiki.resource.servlet.ResourceReferenceHandlerServlet.handleResourceReference(ResourceReferenceHandlerServlet.java:159)
	at org.xwiki.resource.servlet.ResourceReferenceHandlerServlet.service(ResourceReferenceHandlerServlet.java:87)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:860)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
	at org.xwiki.container.servlet.filters.internal.SetHTTPHeaderFilter.doFilter(SetHTTPHeaderFilter.java:63)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:566)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:163)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:73)
	at org.xwiki.resource.servlet.RoutingFilter.doFilter(RoutingFilter.java:147)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1629)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:530)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.eclipse.jetty.io.EofException: null
	at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:292)
	at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:429)
	at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:322)
	at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:372)
	at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:788)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:224)
	at org.eclipse.jetty.server.HttpConnection.send(HttpConnection.java:528)
	at org.eclipse.jetty.server.HttpChannel.sendResponse(HttpChannel.java:778)
	at org.eclipse.jetty.server.HttpChannel.write(HttpChannel.java:834)
	at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:234)
	at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:218)
	at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:509)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2315)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:2270)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2291)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:2246)
	at org.xwiki.resource.servlet.AbstractServletResourceReferenceHandler.serveResource(AbstractServletResourceReferenceHandler.java:185)
	... 58 common frames omitted
Caused by: java.io.IOException: Broken pipe
	at sun.nio.ch.FileDispatcherImpl.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:51)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:270)
	... 75 common frames omitted
`

**Comment by mflorea** [comment]
@CamiAndrei I cannot reproduce with XWiki 8.4 and Diagram Pro 1.10 on Firefox, no matter how many times I reload the page. It would help if you can check the JavaScript console when this happens again. The stacktrace you pasted doesn't seem related (at least it shouldn't prevent the diagram from being displayed).

Would be great if you can re-try with Diagram Pro 1.11. Thanks.

**Comment by CamiAndrei** [comment]
I have tested again this with version 10.0 .  I have reproduced only with empty diagram pages. 
* Create a diagram page and click save and view before the editor loaded. 
After I upgraded the extension to 10.11, the previous diagram didn't loaded anymore: 
![Screenshot from 2019-11-25 14-06-06](https://user-images.githubusercontent.com/12369283/69539449-2d1e7e80-0f8d-11ea-9469-be054e00142f.png)
It doesn't reproduce on 10.11 fresh installed.
These tests were made with XWiki 8.4 and 9.11.3, browser: Mozilla Firefox 70.0.1. On Chrome 78.0.3904.108 seems to work just fine.
